src/utils/generator.js

```diff
@@ -94,15 +243,65 @@ export class Generator {
     this._entries = [];
     this.id = realm.nextGeneratorId++;
     this._name = name;
+    this.effectsToApply = effects;
   }
 
   realm: Realm;
   _entries: Array<GeneratorEntry>;
   preludeGenerator: PreludeGenerator;
+  effectsToApply: void | Effects;
 
   id: number;
   _name: string;
 
+  // Make sure to to fixup
+  // how to apply things around sets of things
+  static fromEffects(effects: Effects, realm: Realm): Generator {
+    let [result, generator, modifiedBindings, modifiedProperties, createdObjects] = effects;
+
+    let output = new Generator(realm, "AdditionalFunctionEffects" + additionalFunctionUID++, effects);
```

Comment:
The generator name doesn't have to be unique, and there's already a globally unique id created for generators anyone, so I would just skip the `additionalFunctionUID` part.